### PR TITLE
add docs infra

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

**What this PR does / why we need it**:
/kind feature

setup docs infra build by [Hugo](https://gohugo.io/) and [Docsy](https://www.docsy.dev/)

with this we can:
- shared markdown snippets cross files mostly like istio.io do
- write docs WYSIWYG by running `hugo run -p 3000 --bind 0.0.0.0`

image

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/REAMDME.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Kurator
```
Comment (hzxuzhonghu) [inline docs/REAMDME.md:1]: file name typo

File: docs/go.mod [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module kurator.dev/kurator/docs
+
+go 1.18
```
Comment (hzxuzhonghu) [inline docs/go.mod:3]: should we keep same a code

File: docs/REAMDME.md (hunk 1 above)
Comment (zirain) [reply docs/REAMDME.md]: fixed

File: docs/go.mod (hunk 2 above)
Comment (zirain) [reply docs/go.mod]: yes, fixed

File: docs/README.md [hunk 3]
```diff
@@ -1,9 +1,8 @@
 # Kurator
 
-## Overview
+Kurator doc is build by [Hugo](https://gohugo.io/) and [Docsy](https://www.docsy.dev/)
```
Comment (hzxuzhonghu) [inline docs/README.md:3]: built

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by prodanlabs** [comment]
Sorry, I only see it now.

looks good to me.

**Pull request merged**